experimental/dds/tree2/src/shared-tree-core/editManager.ts

```diff
@@ -536,9 +525,9 @@ export class EditManager<
 		this.trunk = mintTrunkCommit(this.trunk, commit, sequenceNumber);
 		if (local) {
 			const type =
-				this.localCommitTypes.get(this.trunk.revision) ??
+				this.localBranchUndoRedoManager.commitTypes.get(this.trunk.revision) ??
 				fail("Local commit types must be tracked until they are sequenced.");
-			this.localCommitTypes.delete(this.trunk.revision);
+			this.localBranchUndoRedoManager.commitTypes.delete(this.trunk.revision);
```

Comment:
It's not very clear what the invariants/responsibilities are around `commitTypes`. `commitTypes` is updated privately by URM for new commits, but then is expected to have entries deleted from outside of the URM. When is it safe to delete? Do you have to delete? What happens if you don't delete? Why should you delete? Answering all these questions is the burden of the _consumer_ of URM, but URM needs to be the thing that answers these questions (via spec/doc/contract) because it owns the data. Given the current documentation, if somebody who hadn't used URM before wanted to use it in, say, a unit test, I think they'd have a hard time understanding what this `commitTypes` thing is that they are passing in, what to do with it, and when. 

It seems to me that `commitTypes` is serving two purposes right now:
* It makes it easy for URM to remember which commits have which types during merge and rebase
* It makes it easy for the EditManager to remember the type of sequenced commits

Considering just the second concern, I think it makes sense to delete entries in this way. We know that after sequencing, the EditManager will never need to know the commit type again. But since this map is shared among all the URMs, is it safe to delete WRT the non-trunk URMs? What happens if there's some branch which has some commits with the same revision tags as some commits on the local branch, and then the ones on the local branch gets sequenced? Won't the URM on that lower branch no longer be able to rebase properly, because the entries got deleted from its map?

Reply:
These are great points. To answer the question about delete, this is a mistake and when cloning, the map should also be cloned so that only the local URM's `commitTypes` ever gets deleted from. You also don't have to delete, it's just a premature optimization. The consumer isn't actually expected to ever interact with `commitTypes` but the local and trunk URMs have a special relationship that allows us to do some clean up.

If this is confusing though, we could just make `commitTypes` private and not clean up.

Reply:
Discussed offline and we decided it would be best to have all URMs share the map. Entries are removed when they are trimmed from the trunk.